app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,34 @@
+class GroupsController < ApplicationController
+  before_action :authenticate_user!, except: [:index]
+
+  def index
+    return unless current_user
+
+    @user = current_user
+    @groups = @user.groups.includes(:entities)
+  end
+
+  def show
+    @group = Group.find(params[:id])
+    @entities = @group.entities.order('created_at ASC')
```

Comment:
- [ ] The entities should be ordered by the most recent ones. Kindly fix this. You should order by 'DESC', not 'ASC'